pkg/segment/query/processor/inputlookupcommand.go

```diff
@@ -0,0 +1,174 @@
+// Copyright (c) 2021-2024 SigScalr, Inc.
+//
+// This file is part of SigLens Observability Solution
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+package processor
+
+import (
+	"compress/gzip"
+	"encoding/csv"
+	"fmt"
+	"io"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/siglens/siglens/pkg/config"
+	"github.com/siglens/siglens/pkg/segment/query/iqr"
+	"github.com/siglens/siglens/pkg/segment/structs"
+	"github.com/siglens/siglens/pkg/segment/utils"
+	putils "github.com/siglens/siglens/pkg/utils"
+)
+
+type inputlookupProcessor struct {
+	options   *structs.InputLookup
+	eof       bool
+	qid       uint64
+	start     uint64
+	processed uint64
+}
+
+func checkCSVFormat(filename string) bool {
+	return strings.HasSuffix(filename, ".csv") || strings.HasSuffix(filename, ".csv.gz")
+}
+
+func createRecord(columnNames []string, record []string) (map[string]utils.CValueEnclosure, error) {
+	if len(columnNames) != len(record) {
+		return nil, fmt.Errorf("createRecord: Column and record lengths are not equal, columnNames: %v, record: %v", columnNames, record)
+	}
+	recordMap := make(map[string]utils.CValueEnclosure)
+	for i, col := range columnNames {
```

Comment:
nit: typically whenever you are refrerring to columnname, be explicit about the var name, col is too generic and it can mean a few different things. 
In most of the code base we use something like 
`for, i, cname := range columnNames`

Reply:
addressed in the latest commit.